pkg/webhook/resources/version/validator.go

```diff
@@ -55,5 +60,25 @@ func (v *versionValidator) checkAnnotations(version *v1beta1.Version) error {
 
 func (v *versionValidator) Update(_ *types.Request, _ runtime.Object, newObj runtime.Object) error {
 	newVersion := newObj.(*v1beta1.Version)
-	return v.checkAnnotations(newVersion)
+	return checkVersion(newVersion)
+}
+
+func checkVersion(version *v1beta1.Version) error {
+	if err := checkAnnotations(version); err != nil {
+		return err
+	}
+	if err := checkISOChecksum(version); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkISOChecksum(version *v1beta1.Version) error {
+	isoChecksum := version.Spec.ISOChecksum
+	// if an isoChecksum is provided, it must be in the SHA-512 format
+	// since Longhorn backing images only accept hashes in that format
+	if isoChecksum != "" && !SHA512Pattern.MatchString(isoChecksum) {
```

Comment:
```
There is SHA256, SHA384 and SHA512. There's no SHA1024 yet.
```

Someone mentioned in another issue, he only got SHA256 and needed to compute a SHA512 manually.

If we force it to be SHA512 format, we need to document it.

Reply:
Sure, I'll file another issue about the documentation.

Reply:
gentle ping @w13915984028 , I've checked the documentation and found that we already mentioned that version CR should use SHA-512 checksum in https://docs.harvesterhci.io/v1.4/upgrade/index#prepare-an-air-gapped-upgrade
```
    isoChecksum: <SHA-512 checksum of the ISO>
```